templates/locals.tftpl

```diff
@@ -1,4 +1,7 @@
 locals {
-  default_location         = location
-  subsciption_logical_name = split("-", "${subscription_name}")[1]
+  default_location         = "${location}"
+  subscription_name         = data.azurerm_subscription.current.display_name
+  subsciption_logical_stage = split("-", local.subscription_name)[0]
+  subsciption_logical_name  = split("-", local.subscription_name)[1]
+  subsciption_logical_num   = split("-", local.subscription_name)[2]
```

Comment:
Resolved with recommendation and also added application_num  = "${split("-",subscription_name)[2]}", because we need to parametrice the code, and write less values in code..